service/history/task/task.go

```diff
@@ -385,9 +405,16 @@ func (t *taskBase) Ack() {
 		t.scope.RecordTimer(metrics.TaskLatencyPerDomain, time.Since(t.submitTime))
 		t.scope.RecordTimer(metrics.TaskQueueLatencyPerDomain, time.Since(t.GetVisibilityTimestamp()))
 	}
+
+	if t.eventLogger != nil && t.shouldProcessTask && t.attempt != 0 {
+		// only dump events when the task should be processed and has been retried
+		t.eventLogger.FlushEvents("Task processing events")
```

Comment:
Could it happen if the task always Nack?

Reply:
Assumption here is all submitted tasks will eventually be acked, otherwise two possible situations
1. task may be lost during the processing and redispatching logic -> ack level won't move -> alert for task lag will fire
2. task got stuck and keep retrying, alert for task attempts will fire and the reason why it's keep retrying will be logged.

The main purpose for this events is for the case that neither of the above two situations occurs, but a task still appears to be lost or not processed. In that case we can be sure that a task is acked, and we can check its life cycle events.